animation_nodes/nodes/color/c_utils.pyx

```diff
@@ -0,0 +1,27 @@
+# cython: profile=True
+cimport cython
+
+from ... data_structures cimport (
+    ColorList,
+    VirtualColorList,
+    PolygonIndicesList
+)
+
+def loopsColorsList(long Amount, PolygonIndicesList polygons, VirtualColorList colors, 
+                    ColorList loopsColors):
+    cdef long i
+    for i in range(Amount):
+        loopsColors[i] = colors[polygons.indices[i]]
+    return loopsColors
+    
+def polygonsColorsList(long Amount, PolygonIndicesList polygons, VirtualColorList colors,
+                       ColorList polygonsColors):
+    cdef long i, j, index
+    cdef long polygonsAmount = len(polygons)
+    index = 0    
+    for i in range(polygonsAmount):
+        for j in range(len(polygons[i])):
+            polygonsColors[index] = colors[i]
+            index += 1
```

Comment:
I think you can just use the `polyLengths` list of polygons in this case.

Reply:
Now, it looks like this,
```
def getLoopColorsFromPolygonColors(PolygonIndicesList polygons, VirtualColorList colors):
    cdef long i, j, index
    cdef ColorList loopsColors = ColorList(length = len(polygons.indices))
    
    index = 0 
    for i in range(len(polygons)):
        for j in range(polygons.polyLengths[i]):
            loopsColors[index] = colors[i]
            index += 1
    return loopsColors
```

Reply:
Looks correct.